### PR TITLE
Range based sampling

### PR DESCRIPTION
**Pull request by ulbg**
Adds options to specify ranges for steps/cfg scale/denoise. A random number will be selected between the ranges to be used for sampling. Useful for experimenting and then refining a specific picture.

### CODE REVIEW

File: webui.py [hunk 1]
```diff
@@ -900,22 +916,55 @@ def process_images(
...
     skip_save = 2 not in toggles
     skip_grid = 3 not in toggles
     sort_samples = 4 in toggles
     write_info_files = 5 in toggles
     jpg_sample = 6 in toggles
-    use_GFPGAN = 7 in toggles
-    use_RealESRGAN = 8 in toggles
+    range_based_gen = 7 in toggles
+    use_GFPGAN = 8 in toggles
+    use_RealESRGAN = 8 in toggles if GFPGAN is None else 9 in toggles # possible index shift
```
Comment (hlky) [inline webui.py:967]: Just change the numbers, there's no need to have if else, I keep removing it and you (or someone else without me noticing on merge) keeps adding it back in

File: webui.py [hunk 2]
```diff
@@ -1042,8 +1107,9 @@ def img2img(prompt: str, image_editor_mode: str, init_info, mask_mode: str, mask
     sort_samples = 6 in toggles
     write_info_files = 7 in toggles
     jpg_sample = 8 in toggles
-    use_GFPGAN = 9 in toggles
-    use_RealESRGAN = 10 in toggles
+    range_based_gen = 9 in toggles
+    use_GFPGAN = 10 in toggles
+    use_RealESRGAN = 10 in toggles if GFPGAN is None else 11 in toggles # possible index shift
```
Comment (hlky) [inline webui.py:1112]: same as above
remember: if in doubt, don't mess, you could have just added your option to the end

File: webui.py [hunk 3]
```diff
@@ -1077,8 +1143,7 @@ def img2img(prompt: str, image_editor_mode: str, init_info, mask_mode: str, mask
         keep_mask = False
 
     assert 0. <= denoising_strength <= 1., 'can only work with strength in [0.0, 1.0]'
-    t_enc = int(denoising_strength * ddim_steps)
```
Comment (hlky) [inline webui.py:1080]: why have you moved this? pretty sure this will break non k-diffusion samplers which need t_enc too

File: webui.py [hunk 4]
```diff
@@ -1102,28 +1167,28 @@ def init():
 
         return init_latent,
 
-    def sample(init_data, x, conditioning, unconditional_conditioning, sampler_name):
+    def sample(init_data, x, conditioning, unconditional_conditioning, sampler_name, steps, cfgs, denoise_str):
         if sampler_name != 'DDIM':
             x0, = init_data
+            t_enc = int(denoise_str * steps)
 
-            sigmas = sampler.model_wrap.get_sigmas(ddim_steps)
```
Comment (hlky) [inline webui.py:1109]: why have you changed this?

**Review by hlky — CHANGES_REQUESTED** [review]
honestly this seems like so much bloat and potential for issues for such a niche feature it's not worth merging
it's just a random value too, can't you pick one yourself? if it was a loop that perhaps tried all cfg values in the range in the same 0.5 steps as the ui to try and recover the exact settings to recreate an image with known seed and prompt or something a bit more useful i would be more inclined to merge

**Comment by ulbg** [comment]
Found it pretty useful for getting the image that I want but it's fine, I will close this.

**Pull request closed without merging**